### ISSUE TITLE
c-column in tabular not always centered

### ISSUE THREAD

**Issue by FrankMittelbach** [latex3/latex2e#102]
## Brief outline of the bug

If some command does an \unskip it removes the \hfil of a c-column

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{article}

% this tries to be invisible
\newcommand\beinvisible[1]{\leavevmode\unskip\strut\ignorespaces}

\begin{document}

\begin{tabular}{clr}
 AA & BB & CC \\
 x  &  y & z  \\
 \beinvisible{} x & \beinvisible{} y & \beinvisible{} z \\
\end{tabular}

\end{document}
```

## Log file (required) and possibly PDF file  
[gh102.log](https://github.com/latex3/latex2e/files/2716785/gh102.log)
[gh102.pdf](https://github.com/latex3/latex2e/files/2716786/gh102.pdf)

**Comment by FrankMittelbach** [comment]
I guess one can argue if this is a bug or a too simple-minded approach but such commands exist that try to remove a preceding space with \unskip. Other parts of tabular have \hskip1sp to guard against this type of case (or rather to guard against something else but ...) so I think the c columns should get the same treatment.

**Comment by FrankMittelbach** [comment]
resolved by `[697b469](/latex3/latex2e/commit/697b469a4e2ca84d72f3c28c62bb5378c6dc0ffb) `
